### PR TITLE
Example of a site with automatic http to https redirects

### PR DESCRIPTION
**Pull request by cquiroz**
I had to augment my main project to automatically redirect http to https and I thought it would be a good example for http4s

Do you think this could be of interest in general? Should I add some documentation?

### CODE REVIEW

File: examples/src/main/scala/com/example/http4s/ssl/SslExampleWithRedirect.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  // TODO: Reference server.jks from something other than one child down.
+  val keypath = Paths.get("../server.jks").toAbsolutePath().toString()
+
+  def builder: ServerBuilder with SSLKeyStoreSupport
+
+  val redirectService = HttpService {
+    case request =>
+      request.headers.get(Host) match {
+        case Some(Host(host, _)) =>
```
Comment (rossabaker) [inline examples/src/main/scala/com/example/http4s/ssl/SslExampleWithRedirect.scala]: This won't work with a request that passes an absolute URI.  Maybe `baseUri = Uri.copy(scheme = "https".ci, port = securePort)`?

**Comment by cquiroz** [comment]
How would it be a middleware? That could be the case if you are in the same port and you get an http request instead of https?

**Comment by rossabaker** [comment]
Oops, you're right.  It wouldn't call a wrapped service unless you were abusing ports.

**Comment by cquiroz** [comment]
Updated as suggested and rebased

**Pull request merged**